sheerwater_benchmarking/data/tahmo.py

```diff
@@ -0,0 +1,183 @@
+"""Get Tahmo data."""
+import dask.dataframe as dd
+import xarray as xr
+
+from sheerwater_benchmarking.utils.caching import cacheable
+from sheerwater_benchmarking.utils import get_grid_ds, get_grid, roll_and_agg, apply_mask, clip_region
+from sheerwater_benchmarking.utils.remote import dask_remote
+
+@dask_remote
+@cacheable(data_type='tabular', cache_args=['station_code'], cache=True)
+def tahmo_station(station_code):
+    """Get Tahmo station data."""
+    obs = dd.read_csv(f"gs://sheerwater-datalake/tahmo-data/tahmo_qc_stations_v2/{station_code}.csv",
+                      on_bad_lines="skip")
+
+    # Get a list of columns starting with quality and merge them
+    quality_cols = [col for col in obs.columns if col.startswith('quality')]
+    obs['qual'] = obs[quality_cols].mean(axis=1)
+
+    # Get a list of columns starting with pr and merge them
+    pr_cols = [col for col in obs.columns if col.startswith('pr')]
+    obs['precip'] = obs[pr_cols].mean(axis=1)
+    obs['station_code'] = station_code
+
+    # Drop the quality columns
+    obs = obs.drop(columns=quality_cols)
+    obs = obs.drop(columns=pr_cols)
+
+    return obs
+
+
+@dask_remote
+@cacheable(data_type='array', cache_args=['grid', 'cell_aggregation'],
+           timeseries='time',
+           chunking={
+               'time': 365,
+               'lat': 300,
+               'lon': 300,
+        }, validate_cache_timeseries=False)
+def tahmo_raw(start_time, end_time, grid='global0_25', cell_aggregation='first'): # noqa: ARG001
+    """Get tahmo data from the QC controlled stations."""
+    # Get the station list
+    stat = dd.read_csv("gs://sheerwater-datalake/tahmo-data/tahmo_station_locs.csv")
+    stations = stat['station code'].unique()
+
+    files = []
+    for station in stations:
+        try:
+            files.append(tahmo_station(station, backend='parquet', filepath_only=True))
+        except FileNotFoundError:
+            print(f"File not found for station {station}. Skipping")
+
+    print(files[0].split('/')[:-1])
+    path = '/'.join(files[0].split('/')[:-1])
+    obs = dd.read_parquet(path)
```

Comment:
Can you add a comment about what's doing on here? You get all the files, join them, and then to a multi-file read in parquet